audio-worklet/test/bitcrusher-test.js

```diff
@@ -0,0 +1,148 @@
+  bitcrusher.processBuffer_(factor, 24, b, outputB);
+
+  let output = outputA.concat(outputB);
+
+  output.map((data, index) => {
+    let expected = Math.floor(index / factor) * factor;
+    console.assert(
+        data === expected, 'computed ' + data + ' but expected ' + expected);
+  });
+}
+
+/**
+ * Create one oscillator, and connect it to a bitcrusher with variables set
+ * to avoid any effect and a delay node which compensates for the script
+ * processor latency. Then verify that the samples are nearly identical.
+ */
+function testScriptProcessor() { 
+  let context = new OfflineAudioContext(2, 44100, 44100);
+
+  const reduction = 1;
+  const bitDepth = 24;
+  const bufferLength = 512;
+  let oscillator = new OscillatorNode(context);
+  oscillator.start();
+
+  let merger = new ChannelMergerNode(context, {numberOfInputs: 2});
+  let bitcrusher = new Bitcrusher(context, {
+    buffersize: bufferLength,
+    inputChannels: 1,
+    outputChannels: 1,
+    bitDepth: bitDepth,
+    reduction: reduction
+  });
+
+  // Accomodate for script processor latency by delaying the oscillator.
+  let delay = context.createDelay();
+  delay.delayTime.value = bufferLength / context.sampleRate;
+
+  oscillator.connect(bitcrusher.input);
+  bitcrusher.output.connect(merger, 0, 0);
+  oscillator.connect(delay).connect(merger, 0, 1);
+  merger.connect(context.destination);
+
+  // When audio buffer is ready, verify bitcrushed samples are unaltered.
+  context.startRendering()
+      .then((buffer) => {
+        let bitcrusherOutput = buffer.getChannelData(0);
+        let originalOutput = buffer.getChannelData(1);
+
+        // Allow for fractional error beyond audible perception. This error
+        // occurs because any sample passing through bitcrusher will undergo
+        // manipulation in Math.round() and will therefore be represented by a
+        // new floating point number that differs slightly from the original.
+        // In tested samples and at a sample rate of 44,1k, the maximum observed
+        // error is 6.183981895446777e-7
+        const permittedSampleError = 6.1840e-7;
+
+        // Verify samples from unadultered oscillator match samples from
+        // bitcrushed oscillator with non-information reducing parameters.
+        for (let i = 0; i < originalOutput.length; i++) {
+          let crushedSample = bitcrusherOutput[i];
+          let originalSample = originalOutput[i];
+          const diff = Math.abs(originalSample - crushedSample);
+          
+          if (i < bufferLength) {
+            console.assert(
+                bitcrusherOutput[i] == 0,
+                'Bitcrusher sample at ' + i + 'expected to be 0 but it was ' +
+                    bitcrusherOutput[i]);
+          }
+
+          console.assert(
+              diff < permittedSampleError,
+              'Bitcrushed sample at ' + i + ' is ' + crushedSample + ' but ' +
+                  originalSample + ' for oscillator');
+        }
+      });
```

Comment:
This code doesn't lend itself for easy extension or update.  You stop on the first failure.  This makes it really difficult to update your thresholds if things change.

What you should do:  Got through the array and find the max error.  Then test the max error against the permitted error.  This makes it easier to update the thresholds if needed because you've computed the max error.  You can still print out the location and values of the first error if you want.